src/backend/opencl/device_manager.hpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************
+ * Copyright (c) 2019, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <common/MemoryManager.hpp>
+#include <platform.hpp>
+
+// Forward declaration from clFFT.h
+struct clfftSetupData_;
+using clfftSetupData = clfftSetupData_;
```

Comment:
Are these supposed to be defined in the global namespace?

Reply:
May be the `using` can be moved inside `DeviceManager` class. But the the other forward declaration is from clfft. We can't move it.